dhis-2/dhis-support/dhis-support-cache-invalidation/src/main/java/org/hisp/dhis/cacheinvalidation/DbChangeEventHandler.java

```diff
@@ -164,30 +192,90 @@ private void tryEvictCache( RecordChangeEvent<SourceRecord> event )
             return;
         }
 
-        Serializable entityId = getEntityId( sourceRecord );
-        Objects.requireNonNull( entityId, "Failed to extract entity id!" );
+        Class<?> firstEntityClass = (Class<?>) entityClasses.get( 0 )[0];
+
+        Serializable entityId = null;
+
+        if ( firstEntityClass.equals( DataValue.class ) )
```

Comment:
A `Class` comparison should either use `==` or `Class#isAssignableFrom`

Reply:
Thanks